mne/channels/montage.py

```diff
@@ -684,12 +722,9 @@ def _set_montage(info, montage, update_ch_names=False):
                  'left untouched.')
 
     elif isinstance(montage, DigMontage):
-        dig = _make_dig_points(nasion=montage.nasion, lpa=montage.lpa,
-                               rpa=montage.rpa, hpi=montage.hpi,
-                               dig_points=montage.hsp,
-                               dig_ch_pos=montage.dig_ch_pos)
-        info['dig'] = dig
-        info['dev_head_t']['trans'] = montage.dev_head_t
+        info['dig'] = montage._get_dig()
+        if montage.dev_head_t is not None:
+            info['dev_head_t']['trans'] = montage.dev_head_t
         if montage.dig_ch_pos is not None:  # update channel positions, too
```

Comment:
The assumption AFAIK is that MEG channels are not in these montages, so they shouldn't be in device coords but rather unknown or head coords, depending on whether or not they have been transformed.